scrc/dataset_creation/pretraining_dataset_creator.py

```diff
@@ -27,27 +27,15 @@ def __init__(self, config: dict):
         self.dataset_name = "swiss_caselaw"
         self.feature_cols = [Section.FULL_TEXT]
 
-    def prepare_dataset(self, save_reports) -> datasets.Dataset:
+    def prepare_dataset(self, save_reports, court_string) -> datasets.Dataset:
         engine = self.get_engine(self.db_scrc)
-        court_strings = next(self.select(engine, "court", "court_string", None))["court_string"].tolist()
-        hf_dataset = datasets.Dataset.from_pandas(pd.DataFrame())  # init empty dataset
-
         data_to_load = {
             "section": True, "file": False, "file_number": False,
             "judgment": False, "citation": False, "lower_court": False
         }
-
-        for court_string in court_strings:
-            # we don't use the cache since it is overwritten after each court
-            df = self.get_df(engine, data_to_load, court_string=court_string, use_cache=False)
-
-            print_memory_usage([df, hf_dataset])
-
-            self.logger.info("Concatenating datasets")
-            hf_dataset = concatenate_datasets([hf_dataset, datasets.Dataset.from_pandas(df)])
-
-            del df
-            gc.collect()
+        # we don't use the cache since it is overwritten after each court
+        df = self.get_df(engine, data_to_load, court_string=court_string, use_cache=False)
```

Comment:
I guess it makes sense to set the use_cache flag based on the concatenate flag, right? So if concatenate is True, use_cache is false

Reply:
Only in the pretraining? I remember I got always the same dataset, when trying to create different datasets and use_cache was True but I don't know if this is still the case.